packages/core/src/modules/job/job-repository.ts

```diff
@@ -0,0 +1,93 @@
+import type { DBClient } from "@dotkomonline/db"
+import type { Job, JobId, JobWrite } from "@dotkomonline/types"
+import type { AttendeeService } from "../attendance/attendee-service"
+import { JobHandlerNotFound } from "./job-error"
+import { AttemptReserveAttendeeJob } from "./jobs/attempt-reserve-attendee-job"
+import type { AnyJob } from "./jobs/generic-job"
+
+export interface JobRepository {
+  create(data: JobWrite): Promise<AnyJob>
+  getById(id: JobId): Promise<AnyJob | null>
+  getEnabledJobs(): Promise<AnyJob[]>
+  getAllJobs(): Promise<AnyJob[]>
+  update(id: string, data: Partial<JobWrite>): Promise<AnyJob>
+  delete(id: JobId): Promise<void>
+}
+
+export class JobsRepositoryImpl implements JobRepository {
+  private readonly db: DBClient
+
+  private readonly attendeeService: AttendeeService
+
+  constructor(db: DBClient, attendeeService: AttendeeService) {
+    this.db = db
+    this.attendeeService = attendeeService
+  }
+
+  private parse(data: Job): AnyJob {
+    switch (data.handlerName) {
```

Comment:
Just a nit, but I would prefer this be named jobName, handler is quite ambiguous